src/GafferSceneUI/LightTool.cpp

```diff
@@ -1394,59 +1356,313 @@ class SpotLightHandle : public LightToolHandle
 					)
 				);
 
-				if( currentSolidArc )
-				{
-					ToGLMeshConverterPtr meshConverter = new ToGLMeshConverter( currentSolidArc );
-					solidAngleGroup->addChild( runTimeCast<IECoreGL::Renderable>( meshConverter->convert() ) );
-				}
-				if( previousSolidArc )
+				if( currentSolidArc )
+				{
+					ToGLMeshConverterPtr meshConverter = new ToGLMeshConverter( currentSolidArc );
+					solidAngleGroup->addChild( runTimeCast<IECoreGL::Renderable>( meshConverter->convert() ) );
+				}
+				if( previousSolidArc )
+				{
+					ToGLMeshConverterPtr meshConverter = new ToGLMeshConverter( previousSolidArc );
+					solidAngleGroup->addChild( runTimeCast<IECoreGL::Renderable>( meshConverter->convert() ) );
+				}
+
+				rootGroup->addChild( solidAngleGroup );
+			}
+
+			rootGroup->setTransform( handleTransform );
+		}
+
+		std::vector<const Inspector::Result *> handleValueInspections()
+		{
+			std::vector<const Inspector::Result *> result;
+			for( const auto &i : inspections() )
+			{
+				for( const auto p: i )
+				{
+					if( p.first == ( m_handleType == HandleType::Cone ? g_coneAngleParameter : g_penumbraAngleParameter ) )
+					{
+						result.push_back( p.second.inspection.get() );
+					}
+				}
+			}
+
+			return result;
+		}
+
+		std::string tipPlugSuffix() const override
+		{
+			return m_handleType == HandleType::Cone ? "cone angles" : "penumbra angles";
+		}
+
+	private :
+
+		void setScenePathInternal( ScenePathPtr scenePath ) override
+		{
+			ConstCompoundObjectPtr attributes = getScenePath()->getScene()->fullAttributes( getScenePath()->names() );
```

Comment:
I think we need to be scoping a context here? Might be worth checking the other places we access the scene too - maybe worth updating your test scene to include a script-level context variable, and reference it from the graph in such a way that errors are thrown if it's not scoped?

Reply:
We don't need to scope a path for `fullAttributes()`, do we?

Reply:
Not a path, no, because that is dealt with inside `fullAttributes()`. I was thinking about other variables that might be in the context...

Reply:
Right, I see now. Though it was motivated by setting the path, I am scoping a context at https://github.com/ericmehl/gaffer/blob/98d314c1bbc7aaa560a05047f1f80757f312d7be/src/GafferSceneUI/LightTool.cpp#L2620 which covers the `updateHandlePath()` and subsequent `handlePathChanged()` call.

Reply:
But I also added a test to make sure that doesn't get lost.